config/registry_image_pruner/image_pruner/prune_images.py

```diff
@@ -13,41 +19,87 @@
     for image in images:
-        # attribute or sbom image
+        # attestation or sbom image
         if image.endswith(".att") or image.endswith(".sbom"):
             sha = re.search("sha256-(.*)(.sbom|.att)", image).group(1)
             if f"sha256:{sha}" not in image_digests:
                 if dry_run:
-                    LOGGER.info("Image %s from %s should be removed", image, repository)
+                    LOGGER.info("Image %s from %s/%s should be removed", image, namespace, name)
                 else:
-                    LOGGER.info("Removing image %s from %s", image, repository)
-                    delete_image_endpoint = (
-                        f"{QUAY_API_URL}/repository/{repository}/tag/{image}"
-                    )
-                    response = session.delete(delete_image_endpoint)
-                    response.raise_for_status()
+                    LOGGER.info("Removing image %s from %s/%s", image, namespace, name)
+                    delete_image_repo(quay_token, namespace, name, image)
+        else:
+            LOGGER.debug("%s is not an image with suffix .att or .sbom", image)
 
 
-def process_repositories(repositories, session, dry_run=False):
+def process_repositories(repos: List[ImageRepo], quay_token: str, dry_run: bool = False) -> None:
     global PROCESSED_REPOS
+    for repo in repos:
+        PROCESSED_REPOS += 1
+        namespace = repo["namespace"]
+        name = repo["name"]
+        LOGGER.info("Processing repository %s: %s/%s", PROCESSED_REPOS, namespace, name)
+        repo_info = get_quay_repo(quay_token, namespace, name)
+        if (images := repo_info.get("tags")) is not None:
+            remove_images(images, quay_token, namespace, name, dry_run=dry_run)
 
-    for repo in repositories:
-        repository = f"{repo['namespace']}/{repo['name']}"
 
-        PROCESSED_REPOS += 1
-        LOGGER.info("Processing repository %s: %s", PROCESSED_REPOS, repository)
+def fetch_image_repos(access_token: str, namespace: str) -> Iterator[List[ImageRepo]]:
+    next_page = None
+    resp: HTTPResponse
+    while True:
+        query_args = {"namespace": namespace}
+        if next_page is not None:
+            query_args["next_page"] = next_page
+
+        api_url = f"{QUAY_API_URL}/repository?{urlencode(query_args)}"
+        request = Request(api_url, headers={
+            "Authorization": f"Bearer {access_token}",
+        })
+
+        with urlopen(request) as resp:
+            if resp.status != 200:
+                raise RuntimeError(resp.reason)
+            json_data = json.loads(resp.read())
 
-        repository_endpoint = f"{QUAY_API_URL}/repository/{repository}"
-        response = session.get(repository_endpoint)
-        response.raise_for_status()
-        repository_json = response.json()
+        repos = json_data.get("repositories", [])
+        if not repos:
+            LOGGER.debug("No image repository is found.")
+            break
```

Comment:
Do we want to `break` when there are no repositories? It looks like the script previously didn't do that

Reply:
I previously chose that approach because it felt more pythonic to me
as the first line of the function 
```
for repo in repos:
```
would just ignore empty repos. This way we have to have a condition and a break, but if the images can really be `None` then this approach make sense